spec-tests/src/test/java/org/candlepin/spec/entitlements/EntitlementResourceSpecTest.java

```diff
@@ -0,0 +1,452 @@
+        ProductDTO product = Products.random();
+        product.setAttributes(List.of(new AttributeDTO().name("multi-entitlement").value("yes")));
+        product = ownerProductApi.createProductByOwner(owner.getKey(), product);
+        PoolDTO pool = Pools.random().productId(product.getId()).quantity(50L);
+        pool = ownerClient.createPool(owner.getKey(), pool);
+
+        List<Thread> threads = new ArrayList<>();
+        List<Pair> data = List.of(new Pair("system", "5"),
+            new Pair("candlepin", "7"),
+            new Pair("system", "6"),
+            new Pair("candlepin", "11"));
+        for (Pair pair : data) {
+            final String poolId = pool.getId();
+            Thread t = new Thread(() -> {
+                try {
+                    registerAndConsume(poolId, pair.getName(), Integer.parseInt(pair.getValue()));
+                }
+                catch (Exception ae) {
+                    throw new RuntimeException(ae);
+                }
+            });
+            threads.add(t);
+        }
+        // Run all the threads
+        for (Thread t : threads) {
+            t.start();
+        }
+        for (Thread t : threads) {
+            t.join();
+        }
+
+        // ensure that the threads finish
+        sleep(15000);
+        PoolDTO thePool = ownerClient.listOwnerPools(owner.getKey()).get(0);
+        assertEquals(29, thePool.getConsumed());
+        assertEquals(18, thePool.getExported());
+    }
+
+    @Test
+    @DisplayName("should not allow over consumption in pool")
+    public void shouldNotAllowOverConsumption() throws Exception {
+        ProductDTO product = Products.random();
+        product.setAttributes(List.of(new AttributeDTO().name("multi-entitlement").value("yes")));
+        product = ownerProductApi.createProductByOwner(owner.getKey(), product);
+        PoolDTO pool = Pools.random().productId(product.getId()).quantity(3L);
+        pool = ownerClient.createPool(owner.getKey(), pool);
+
+        List<Thread> threads = new ArrayList<>();
+        List<Pair> data = List.of(new Pair("system", "1"),
+            new Pair("candlepin", "1"),
+            new Pair("system", "1"),
+            new Pair("candlepin", "1"),
+            new Pair("candlepin", "1"),
+            new Pair("candlepin", "1"));
+        for (Pair pair : data) {
+            final String poolId = pool.getId();
+            Thread t = new Thread(() -> {
+                try {
+                    registerAndConsume(poolId, pair.getName(), Integer.parseInt(pair.getValue()));
+                }
+                catch (Exception ae) {
+                    throw new RuntimeException(ae);
+                }
+            });
+            threads.add(t);
+        }
+        // Run all the threads
+        for (Thread t : threads) {
+            t.start();
+        }
+        for (Thread t : threads) {
+            t.join();
+        }
+
+        // ensure that the threads finish
+        sleep(15000);
```

Comment:
Why do we need such long pauses compared to Ruby? Shouldn't join() handle the waiting?